### PR TITLE
Fix TorrentLeech provider

### PR DESCRIPTION
**Pull request by oreleraki**
Added completely new native provider using JavaScript. This has been added in order to handle the tracker search query, now based upon JSON request.

Categories also been changed and added.
Fixed and added all new numbers to known categories.
Added 3 more categories: Education, Animation and Foreign.

Fix for issue: #36

### CODE REVIEW

File: lib/providers/torrentleech.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			defaultCategory: 'All',
+		});
+	}
+
+	search(query, category) {
+		const url = this.getUrl(category, query);
+		if (url === null) {
+			return undefined;
+		}
+		return request({ url, jar: this.cookieJar, headers: { 'User-Agent': 'curl/7.37.0' } })
```
Comment (JimmyLaurent) [inline lib/providers/torrentleech.js]: is it possible to use "this.request" instead of "request" ?

File: lib/providers/torrentleech.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+				);
+			});
+	}
+
+	getTorrentDetails(torrent) {
+		throw new Error('Not implemented');
+	}
+
+	downloadTorrent() {
+		throw new Error('Not implemented');
```
Comment (JimmyLaurent) [inline lib/providers/torrentleech.js]: Is it difficult to implement those methods ?

Comment (oreleraki) [reply lib/providers/torrentleech.js]: I can implement the `downloadTorrent` method, but concerning `getTorrentDetails` i a bit confused what should be return if any. If i should return the Html, it is a bit funky, to extract using json, but give the user the raw html, also torrentleech have a long information tab for a torrent, should i give the entire tab info, or just part of it ? and if part of it, then what should be must that ?
I haven't seen any example in previous providers so i wasn't sure.
If you want i can implement it as i do best.
Waiting for your reply.

File: lib/providers/torrentleech.js (hunk 1 above)
Comment (oreleraki) [reply lib/providers/torrentleech.js]: Yes, i will

File: lib/providers/torrentleech.js (hunk 2 above)
Comment (JimmyLaurent) [reply lib/providers/torrentleech.js]: Sorry for the delay, I have been working a lot lately.

For the "getTorrentDetails" method, the idea is to get whatever information there is about the torrent. So just do your best ;)

Thanks.

**Pull request merged**